ballerina/channel_csv_io.bal

```diff
@@ -60,35 +60,64 @@ isolated function channelWriteCsv(string path, FileWriteOption option, string[][
 }
 
-isolated function channelWriteCsvFromStream(WritableChannel writableChannel, stream<string[]|map<anydata>, Error?> csvStream) returns
+isolated function channelWriteCsvFromStream(string path, FileWriteOption option, stream<string[]|map<anydata>, Error?> csvStreamToWrite) returns
 Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
-    do {
-        if csvStream is stream<string[], Error?> {
-            record {|string[] value;|}|Error? csvRecordString = csvStream.next();
+    string[] headersFromCSV = [];
+    string[] headersFromStream = [];
+    boolean skipHeaders = false;
+    if csvStreamToWrite is stream<string[], Error?> {
+        WritableCSVChannel csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
+        do {
+            record {|string[] value;|}|Error? csvRecordString = csvStreamToWrite.next();
             while csvRecordString is record {|string[] value;|} {
                 check csvChannel.write(csvRecordString.value);
-                csvRecordString = csvStream.next();
+                csvRecordString = csvStreamToWrite.next();
             }
-        } else if csvStream is stream<map<anydata>, Error?> {
-            record {|map<anydata> value;|}? csvRecordMap = check csvStream.next();
-            while csvRecordMap is record {|map<anydata> value;|} {
-                string[] sValues = [];
-                foreach [string, anydata] [_, value] in csvRecordMap["value"].entries() {
-                    sValues.push(value.toString());
+        } on fail Error err {
+            check csvStreamToWrite.close();
+            check csvChannel.close();
+            return err;
+        }
+        check csvChannel.close();
+    } else if csvStreamToWrite is stream<map<anydata>, Error?> {
+        record {|map<anydata> value;|}? csvRecordMap;
+        if option == OVERWRITE {
+            do {
+                csvRecordMap = check csvStreamToWrite.next();
+                if csvRecordMap !is () {
+                    headersFromStream = csvRecordMap["value"].keys();
+                } else {
+                    check csvStreamToWrite.close();
+                    return;
                 }
-                check csvChannel.write(sValues);
-                csvRecordMap = check csvStream.next();
+                check writeRecordsStreamToCsvFile(path, csvStreamToWrite, headersFromStream, option, skipHeaders, csvRecordMap);
+            } on fail Error err {
+                check csvStreamToWrite.close();
+                return err;
+            }
+        } else {
+            headersFromCSV = check readHeadersFromCsvFile(path);
+            do {
+                csvRecordMap = check csvStreamToWrite.next();
+                if csvRecordMap !is () {
+                    headersFromStream = csvRecordMap["value"].keys();
+                } else {
+                    check csvStreamToWrite.close();
+                    return;
+                }
```

Comment:
These lines are same for both Override and Append

Reply:
updated in [c475f59](https://github.com/ballerina-platform/module-ballerina-io/pull/830/commits/c475f5979f3f0d6e8f7b7b160c9f54d824a33577)